klippy/extras/load_cell_probe.py

```diff
@@ -0,0 +1,699 @@
+        # clear trsync from load_cell_endstop
+        self._home_cmd.send([self._oid, 0, 0, 0, 0, 0, 0, 0])
+        return self._mcu.clock_to_print_time(trigger_ticks)
+
+
+# Execute probing moves using the McuLoadCellProbe
+class LoadCellProbingMove:
+    ERROR_MAP = {
+        mcu.MCU_trsync.REASON_COMMS_TIMEOUT:
+            "Communication timeout during homing",
+        McuLoadCellProbe.ERROR_SAFETY_RANGE:
+            "Load Cell Probe Error: load exceeds safety limit",
+        McuLoadCellProbe.ERROR_OVERFLOW:
+            "Load Cell Probe Error: fixed point math overflow",
+        McuLoadCellProbe.ERROR_WATCHDOG:
+            "Load Cell Probe Error: timed out waiting for sensor data"
+    }
+    def __init__(self, config, mcu_load_cell_probe,
+                 param_helper, continuous_tare_filter_helper, config_helper):
+        self._printer = config.get_printer()
+        self._mcu_load_cell_probe = mcu_load_cell_probe
+        self._param_helper = param_helper
+        self._continuous_tare_filter_helper = continuous_tare_filter_helper
+        self._config_helper = config_helper
+        self._mcu = mcu_load_cell_probe.get_mcu()
+        self._load_cell = mcu_load_cell_probe.get_load_cell()
+        self._z_min_position = probe.lookup_minimum_z(config)
+        self._dispatch = mcu_load_cell_probe.get_dispatch()
+        probe.LookupZSteppers(config, self._dispatch.add_stepper)
+        # internal state tracking
+        self._tare_counts = 0
+        self._last_trigger_time = 0
+
+    def _start_collector(self):
+        toolhead = self._printer.lookup_object('toolhead')
+        # homing uses the toolhead last move time which gets special handling
+        # to significantly buffer print_time if the move queue has drained
+        print_time = toolhead.get_last_move_time()
+        collector = self._load_cell.get_collector()
+        collector.start_collecting(min_time=print_time)
+        return collector
+
+    # pauses for the last move to complete and then
+    # sets the endstop tare value and range
+    def _pause_and_tare(self, gcmd):
+        import numpy as np
+        # update sos_filter with any gcode parameter changes
+        self._continuous_tare_filter_helper.update_from_command(gcmd,
+                                    self._mcu_load_cell_probe.get_cmd_queue())
+        collector = self._start_collector()
+        num_samples = self._config_helper.get_tare_samples(gcmd)
+        # use collect_min collected samples are not wasted
+        results = collector.collect_min(num_samples)
+        tare_samples = check_sensor_errors(results, self._printer)
+        tare_counts = np.average(np.array(tare_samples)[:, 2].astype(float))
+        self._mcu_load_cell_probe.set_endstop_range(tare_counts, gcmd)
+
+    def _home_start(self, print_time):
+        cmd_err = self._printer.command_error
+        if not self._load_cell.is_calibrated():
+            raise cmd_err("Probe aborted, Load cell not calibrated")
```

Comment:
The `phoming.probing_move()` code is not expecting `home_start()` to raise an error.  If something needs to be validated it should be done prior to invoking `phoming.probing_move()`.

Reply:
Ok, deleted.